frontend/src/hooks/useUsers.ts

```diff
@@ -15,3 +16,14 @@ export const useCreateUser = () => {
     isCreateUserLoading,
   };
 };
+
+export const useGetUsername = () => {
+  const getUsername = useCallback(async (userId: number) => {
+    const { data } = await ApiService.get<GetUserNameRes | undefined>(
+      `/users/${userId}`,
+    );
+    return data?.name;
+  }, []);
+
+  return { getUsername };
+};
```

Comment:
Let's use `react-query` to help us cache this:
```suggestion
export const useGetUserName = (userId?: number) => {
  const getUserName = async () => {
    const { data } = await ApiService.get<GetUserNameRes | undefined>(
      `/users/${userId}`,
    );
    return data;
  };
  const { data: user, isLoading: isGetUserNameLoading } = useQuery(
    ["USER", userId?.toString() || ""],
    getUserName,
    {
      enabled: !!userId,
    },
  );
  return {
    user,
    isGetUserNameLoading,
  };
};
```